cloud/ec2_fleet.go

```diff
@@ -209,44 +209,47 @@ func (m *ec2FleetManager) GetInstanceStatuses(ctx context.Context, hosts []host.
 	return statuses, nil
 }
 
-func (m *ec2FleetManager) GetInstanceStatus(ctx context.Context, h *host.Host) (CloudStatus, error) {
-	status := StatusUnknown
+// GetInstanceState returns a universal status code representing the state
+// of an ec2 fleet and a state reason if available. The state reason should not be
+// used to determine the status of the ec2 fleet but rather to provide additional
+// context about the state of the ec2 state.
+func (m *ec2FleetManager) GetInstanceState(ctx context.Context, h *host.Host) (CloudInstanceState, error) {
```

Comment:
Can you please add some more explanation on what a state reason is from the ec2 perspective, perhaps with a link to documentation?

Reply:
Sure!

Reply:
It looks like this isn't done yet but lgtm once complete